src/main/resources/public/ts/models/teacher/ReleveNote.ts

```diff
@@ -578,6 +579,29 @@ export class ReleveNote extends  Model implements IModel {
         });
     }
 
+    private getPreviousPeriod () : Array<Periode> {
+        let previousPeriods : Periode[] = [];
+        previousPeriods = _.filter(this.classe.periodes.all, (p)=> {
+            return p.id != null && p.id_type < this.idPeriode;
+        });
+        return previousPeriods;
+    }
```

Comment:
Tu n'as pas besoin d'utiliser loadash comme on est en es5 on peut utiliser les opération des array de javascript directement

```suggestion
    private getPreviousPeriod(): Array<Periode> {
        let previousPeriods: Periode[];
        previousPeriods = this.classe.periodes.all.filter((p: Periode) => p.id != null && p.id_type < this.idPeriode);
        return previousPeriods;
    }
```